Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyFulfillmentAPI.Codeunit.al

```diff
@@ -6,48 +6,47 @@ namespace Microsoft.Integration.Shopify;
 codeunit 30315 "Shpfy Fulfillment API"
 {
     var
-        ShpfyJsonHelper: Codeunit "Shpfy Json Helper";
         ShpfyCommunicationMgt: Codeunit "Shpfy Communication Mgt.";
 
     /// <summary>
-    /// Fulfills shopify orders for each fulfillment id parsed from a completed draft order.
+    /// Creates a fulfillment for a provided fulfillment order id.
     /// </summary>
-    /// <param name="JResponse">Json response from a completed draft order</param>
-    /// <param name="ShopCode">Shopify shop code to be used.</param>
-    internal procedure FulfillShopifyOrder(JResponse: JsonToken; ShopCode: Code[20])
+    /// <param name="FulfillmentOrderId">Fulfillment order id.</param>
+    internal procedure CreateFulfillment(FulfillmentOrderId: Text)
     var
-        FulfillmentOrderList: List of [Text];
-        FulfillmentOrderId: Text;
-        ResponseJsonToken: JsonToken;
+        JResponse: JsonToken;
         GraphQLType: Enum "Shpfy GraphQL Type";
         Parameters: Dictionary of [Text, Text];
     begin
-        ShpfyCommunicationMgt.SetShop(ShopCode);
-        FulfillmentOrderList := ParseFulfillmentOrders(JResponse);
         GraphQLType := "Shpfy GraphQL Type"::FulfillOrder;
-
-        foreach FulfillmentOrderId in FulfillmentOrderList do begin
-            Parameters.Add('FulfillmentOrderId', FulfillmentOrderId);
-            ResponseJsonToken := ShpfyCommunicationMgt.ExecuteGraphQL(GraphQLType, Parameters);
-            Clear(Parameters);
-        end;
+        Parameters.Add('FulfillmentOrderId', FulfillmentOrderId);
+        JResponse := ShpfyCommunicationMgt.ExecuteGraphQL(GraphQLType, Parameters);
     end;
 
-    local procedure ParseFulfillmentOrders(ResponseJsonToken: JsonToken) FulfillmentOrderList: List of [Text]
+    /// <summary>
+    /// Gets fulfillment orders for a provided shopify order id.
+    /// </summary>
+    /// <param name="OrderId">Shopify order id to get fulfillments from.</param>
+    /// <param name="NumberOfLines">Number of fulfillment orders to get.</param>
+    /// <returns>Fulfillment orders.</returns>
+    internal procedure GetFulfillmentOrders(OrderId: Text; NumberOfLines: Integer) JFulfillments: JsonToken
```

Comment:
I suppose this is okay and we don't have to change it. 

But the usual pattern is that the query has some set amount of nodes it will request, and if there are more nodes, there would be a GetNextFulfillmentOrders. (You can see how GetProudctIds and GetNextProductIds are used)

The reason for this is that it makes the cost of each query more predictable. Right now you're saying that the expected cost is 6. However, if you'll be requesting the first 50 fulfillment orders, because there might be more lines, then the expected cost goes up. But if each query only takes X, and if there's more, it sends a new request, than you can have more predicatbility with resource exhaustion.

But like I said, I don't think it's a big deal right now.